From 123b913175cd83b2acbd37d9f8d4b9fc3dff466b Mon Sep 17 00:00:00 2001
From: WangYuli <wangyuli@aosc.io>
Date: Fri, 5 Dec 2025 12:59:39 +0800
Subject: [PATCH 2/3] src: Add ARM CCA support in domain capabilities command

bugzilla:https://gitee.com/openeuler/libvirt/issues/ICU4UF?from=project-issue
reference:https://patchew.org/Libvirt/20250612071418.2926384-1-fj1078ii@aa.jp.fujitsu.com/

----------------------------------------

- Add ARM CCA support in domain capabilies XML schema.

[Capability example]
- Execution results of 'virsh domcapability" on qemu
<domaincapabilities>
...
  <features>
    ...
    </sgx>
    <cca supported='yes'>
    <enum name='measurement-algo'>
      <value>sha256</value>
      <value>sha512</value>
    </enum>
    </cca>
    <hyperv supported='yes'>
    ...
  </features>
</domaincapabilities>

Signed-off-by: Kazuhiro Abe <fj1078ii@aa.jp.fujitsu.com>
Signed-off-by: rpm-build <rpm-build>
Signed-off-by: WangYuli <wangyuli@aosc.io>
---
 docs/formatdomaincaps.rst                     | 21 +++-
 src/conf/domain_capabilities.c                | 48 +++++++++
 src/conf/domain_capabilities.h                |  6 ++
 src/libvirt_private.syms                      |  1 +
 src/qemu/qemu_capabilities.c                  | 28 ++++++
 src/qemu/qemu_monitor.c                       | 10 ++
 src/qemu/qemu_monitor.h                       |  3 +
 src/qemu/qemu_monitor_json.c                  | 98 +++++++++++++++++++
 src/qemu/qemu_monitor_json.h                  |  4 +
 tests/domaincapsdata/bhyve_basic.x86_64.xml   |  1 +
 tests/domaincapsdata/bhyve_fbuf.x86_64.xml    |  1 +
 tests/domaincapsdata/bhyve_uefi.x86_64.xml    |  1 +
 tests/domaincapsdata/empty.xml                |  1 +
 tests/domaincapsdata/libxl-xenfv.xml          |  1 +
 tests/domaincapsdata/libxl-xenpv.xml          |  1 +
 .../domaincapsdata/qemu_4.2.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_4.2.0-tcg.x86_64.xml  |  1 +
 .../qemu_4.2.0-virt.aarch64.xml               |  1 +
 tests/domaincapsdata/qemu_4.2.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_4.2.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_4.2.0.s390x.xml     |  1 +
 tests/domaincapsdata/qemu_4.2.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_5.0.0-q35.x86_64.xml  |  1 +
 .../qemu_5.0.0-tcg-virt.riscv64.xml           |  1 +
 .../domaincapsdata/qemu_5.0.0-tcg.x86_64.xml  |  1 +
 .../qemu_5.0.0-virt.aarch64.xml               |  1 +
 .../qemu_5.0.0-virt.riscv64.xml               |  1 +
 tests/domaincapsdata/qemu_5.0.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_5.0.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_5.0.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_5.1.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_5.1.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_5.1.0.sparc.xml     |  1 +
 tests/domaincapsdata/qemu_5.1.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_5.2.0-q35.x86_64.xml  |  1 +
 .../qemu_5.2.0-tcg-virt.riscv64.xml           |  1 +
 .../domaincapsdata/qemu_5.2.0-tcg.x86_64.xml  |  1 +
 .../qemu_5.2.0-virt.aarch64.xml               |  1 +
 .../qemu_5.2.0-virt.riscv64.xml               |  1 +
 tests/domaincapsdata/qemu_5.2.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_5.2.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_5.2.0.s390x.xml     |  1 +
 tests/domaincapsdata/qemu_5.2.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_6.0.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_6.0.0-tcg.x86_64.xml  |  1 +
 .../qemu_6.0.0-virt.aarch64.xml               |  1 +
 tests/domaincapsdata/qemu_6.0.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_6.0.0.s390x.xml     |  1 +
 tests/domaincapsdata/qemu_6.0.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_6.1.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_6.1.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_6.1.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_6.2.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_6.2.0-tcg.x86_64.xml  |  1 +
 .../qemu_6.2.0-virt.aarch64.xml               |  1 +
 tests/domaincapsdata/qemu_6.2.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_6.2.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_6.2.0.x86_64.xml    |  1 +
 .../qemu_7.0.0-hvf.aarch64+hvf.xml            |  1 +
 .../domaincapsdata/qemu_7.0.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_7.0.0-tcg.x86_64.xml  |  1 +
 .../qemu_7.0.0-virt.aarch64.xml               |  1 +
 tests/domaincapsdata/qemu_7.0.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_7.0.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_7.0.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_7.1.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_7.1.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_7.1.0.ppc64.xml     |  1 +
 tests/domaincapsdata/qemu_7.1.0.x86_64.xml    |  1 +
 .../qemu_7.2.0-hvf.x86_64+hvf.xml             |  1 +
 .../domaincapsdata/qemu_7.2.0-q35.x86_64.xml  |  1 +
 .../qemu_7.2.0-tcg.x86_64+hvf.xml             |  1 +
 .../domaincapsdata/qemu_7.2.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_7.2.0.ppc.xml       |  1 +
 tests/domaincapsdata/qemu_7.2.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_8.0.0-q35.x86_64.xml  |  1 +
 .../qemu_8.0.0-tcg-virt.riscv64.xml           |  1 +
 .../domaincapsdata/qemu_8.0.0-tcg.x86_64.xml  |  1 +
 .../qemu_8.0.0-virt.riscv64.xml               |  1 +
 tests/domaincapsdata/qemu_8.0.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_8.1.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_8.1.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_8.1.0.s390x.xml     |  1 +
 tests/domaincapsdata/qemu_8.1.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_8.2.0-q35.x86_64.xml  |  1 +
 .../qemu_8.2.0-tcg-virt.loongarch64.xml       |  1 +
 .../domaincapsdata/qemu_8.2.0-tcg.x86_64.xml  |  1 +
 .../qemu_8.2.0-virt.aarch64.xml               |  1 +
 .../qemu_8.2.0-virt.loongarch64.xml           |  1 +
 tests/domaincapsdata/qemu_8.2.0.aarch64.xml   |  1 +
 tests/domaincapsdata/qemu_8.2.0.armv7l.xml    |  1 +
 tests/domaincapsdata/qemu_8.2.0.s390x.xml     |  1 +
 tests/domaincapsdata/qemu_8.2.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_9.0.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_9.0.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_9.0.0.sparc.xml     |  1 +
 tests/domaincapsdata/qemu_9.0.0.x86_64.xml    |  1 +
 .../domaincapsdata/qemu_9.1.0-q35.x86_64.xml  |  1 +
 .../domaincapsdata/qemu_9.1.0-tcg.x86_64.xml  |  1 +
 tests/domaincapsdata/qemu_9.1.0.x86_64.xml    |  1 +
 100 files changed, 309 insertions(+), 1 deletion(-)

diff --git a/docs/formatdomaincaps.rst b/docs/formatdomaincaps.rst
index 712d8b44..935a81a3 100644
--- a/docs/formatdomaincaps.rst
+++ b/docs/formatdomaincaps.rst
@@ -817,6 +817,24 @@ document store. In order to use SGX with libvirt have a look at `SGX in domain X
 ``sections``
    The sections of the SGX enclave page cache (called EPC).
 
+CCA capabilities
+^^^^^^^^^^^^^^^^
+
+Arm Confidential Compute Architecture (CCA) capabilities are exposed under the
+``cca`` element.
+
+Arm CCA is a system solution comprised of hardware and software components that
+maximizes the security of data on devices and in the cloud.
+CCA enhances the virtualization capabilities of the platform by separating the
+management of resources from access to those resources.
+
+For more details on the CCA feature, please follow resources in the CCA developer's
+document store. In order to use CCA with libvirt have a look at `CCA in domain
+XML <formatdomain.html#launch-security>`__
+
+``measurement-algo``
+   Options for the ``measurement-algo`` used to describe blob hashes.
+
 
 Hyper-V Enlightenments
 ^^^^^^^^^^^^^^^^^^^^^^
@@ -838,4 +856,5 @@ The ``sectype`` enum corresponds to ``type`` attribute of ``<launchSecurity/>``
 element as documented in `Launch Security
 <formatdomain.html#launch-security>`__.  :since:`(Since 10.5.0)` For additional
 information on individual types, see sections above: `s390-pv capability`_ for
-S390 PV, `SEV capabilities`_ for AMD SEV and/or AMD SEV-SNP.
+S390 PV, `SEV capabilities`_ for AMD SEV and/or AMD SEV-SNP, `CCA capabilities`_
+for Arm CCA.
diff --git a/src/conf/domain_capabilities.c b/src/conf/domain_capabilities.c
index 30540de3..d104d6e8 100644
--- a/src/conf/domain_capabilities.c
+++ b/src/conf/domain_capabilities.c
@@ -89,6 +89,25 @@ virSGXCapabilitiesFree(virSGXCapability *cap)
 }
 
 
+void
+virCCACapabilitiesFree(virCCACapability *cap)
+{
+    size_t i;
+
+    if (!cap)
+        return;
+
+    if (cap->nCcaMeasurementAlgo)
+        for (i = 0; i < cap->nCcaMeasurementAlgo; i++)
+            g_free(cap->ccaMeasurementAlgo[i]);
+
+    if (cap->ccaMeasurementAlgo)
+        g_free(cap->ccaMeasurementAlgo);
+
+    g_free(cap);
+}
+
+
 static void
 virDomainCapsDispose(void *obj)
 {
@@ -102,6 +121,7 @@ virDomainCapsDispose(void *obj)
     virCPUDefFree(caps->cpu.hostModel);
     virSEVCapabilitiesFree(caps->sev);
     virSGXCapabilitiesFree(caps->sgx);
+    virCCACapabilitiesFree(caps->cca);
     g_free(caps->hyperv);
 
     values = &caps->os.loader.values;
@@ -706,6 +726,33 @@ virDomainCapsFeatureSGXFormat(virBuffer *buf,
     virBufferAddLit(buf, "</sgx>\n");
 }
 
+static void
+virDomainCapsFeatureCCAFormat(virBuffer *buf,
+                              const virCCACapability *cca)
+{
+    size_t i;
+
+    if (!cca) {
+        virBufferAddLit(buf, "<cca supported='no'/>\n");
+        return;
+    }
+
+    virBufferAddLit(buf, "<cca supported='yes'>\n");
+    virBufferAdjustIndent(buf, 2);
+
+    virBufferAddLit(buf, "<enum name='measurement-algo'>\n");
+    virBufferAdjustIndent(buf, 2);
+    for (i = 0; i < cca->nCcaMeasurementAlgo; i++) {
+        virBufferAsprintf(buf, "<value>%s</value>\n",
+                          cca->ccaMeasurementAlgo[i]);
+    }
+    virBufferAdjustIndent(buf, -2);
+    virBufferAddLit(buf, "</enum>\n");
+
+    virBufferAdjustIndent(buf, -2);
+    virBufferAddLit(buf, "</cca>\n");
+}
+
 static void
 virDomainCapsFeatureHypervFormat(virBuffer *buf,
                                  const virDomainCapsFeatureHyperv *hyperv)
@@ -753,6 +800,7 @@ virDomainCapsFormatFeatures(const virDomainCaps *caps,
 
     virDomainCapsFeatureSEVFormat(&childBuf, caps->sev);
     virDomainCapsFeatureSGXFormat(&childBuf, caps->sgx);
+    virDomainCapsFeatureCCAFormat(&childBuf, caps->cca);
     virDomainCapsFeatureHypervFormat(&childBuf, caps->hyperv);
     virDomainCapsLaunchSecurityFormat(&childBuf, &caps->launchSecurity);
 
diff --git a/src/conf/domain_capabilities.h b/src/conf/domain_capabilities.h
index 01076c32..d880d367 100644
--- a/src/conf/domain_capabilities.h
+++ b/src/conf/domain_capabilities.h
@@ -306,6 +306,7 @@ struct _virDomainCaps {
     virDomainCapsFeatureGIC gic;
     virSEVCapability *sev;
     virSGXCapability *sgx;
+    virCCACapability *cca;
     virDomainCapsFeatureHyperv *hyperv;
     virDomainCapsLaunchSecurity launchSecurity;
     /* add new domain features here */
@@ -363,3 +364,8 @@ void
 virSGXCapabilitiesFree(virSGXCapability *capabilities);
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(virSGXCapability, virSGXCapabilitiesFree);
+
+void
+virCCACapabilitiesFree(virCCACapability *capabilities);
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(virCCACapability, virCCACapabilitiesFree);
diff --git a/src/libvirt_private.syms b/src/libvirt_private.syms
index af40e5dc..07b75ca4 100644
--- a/src/libvirt_private.syms
+++ b/src/libvirt_private.syms
@@ -208,6 +208,7 @@ virDomainAuditVcpu;
 
 
 # conf/domain_capabilities.h
+virCCACapabilitiesFree;
 virDomainCapsCPUModelsAdd;
 virDomainCapsCPUModelsCopy;
 virDomainCapsCPUModelsGet;
diff --git a/src/qemu/qemu_capabilities.c b/src/qemu/qemu_capabilities.c
index d3bd015e..16adfa55 100644
--- a/src/qemu/qemu_capabilities.c
+++ b/src/qemu/qemu_capabilities.c
@@ -3539,6 +3539,32 @@ virQEMUCapsProbeQMPSGXCapabilities(virQEMUCaps *qemuCaps,
 }
 
 
+static int
+virQEMUCapsProbeQMPCCACapabilities(virQEMUCaps *qemuCaps,
+                                   qemuMonitor *mon)
+{
+    int rc = -1;
+    virCCACapability *caps = NULL;
+
+    if (!virQEMUCapsGet(qemuCaps, QEMU_CAPS_CCA_GUEST))
+        return 0;
+
+    if ((rc = qemuMonitorGetCCACapabilities(mon, &caps)) < 0)
+        return -1;
+
+    /* CCA isn't actually supported */
+    if (rc == 0) {
+        virQEMUCapsClear(qemuCaps, QEMU_CAPS_CCA_GUEST);
+        return 0;
+    }
+
+    virCCACapabilitiesFree(qemuCaps->ccaCapabilities);
+    qemuCaps->ccaCapabilities = caps;
+    return 0;
+}
+
+
+
 /*
  * Filter for features which should never be passed to QEMU. Either because
  * QEMU never supported them or they were dropped as they never did anything
@@ -5604,6 +5630,8 @@ virQEMUCapsInitQMPMonitor(virQEMUCaps *qemuCaps,
         return -1;
     if (virQEMUCapsProbeQMPSGXCapabilities(qemuCaps, mon) < 0)
         return -1;
+    if (virQEMUCapsProbeQMPCCACapabilities(qemuCaps, mon) < 0)
+        return -1;
 
     virQEMUCapsInitProcessCaps(qemuCaps);
 
diff --git a/src/qemu/qemu_monitor.c b/src/qemu/qemu_monitor.c
index 7f65c237..ed26846f 100644
--- a/src/qemu/qemu_monitor.c
+++ b/src/qemu/qemu_monitor.c
@@ -3442,6 +3442,16 @@ qemuMonitorGetSGXCapabilities(qemuMonitor *mon,
 }
 
 
+int
+qemuMonitorGetCCACapabilities(qemuMonitor *mon,
+                              virCCACapability **capabilities)
+{
+    QEMU_CHECK_MONITOR(mon);
+
+    return qemuMonitorJSONGetCCACapabilities(mon, capabilities);
+}
+
+
 int
 qemuMonitorNBDServerStart(qemuMonitor *mon,
                           const virStorageNetHostDef *server,
diff --git a/src/qemu/qemu_monitor.h b/src/qemu/qemu_monitor.h
index 57d1b45b..86b6e516 100644
--- a/src/qemu/qemu_monitor.h
+++ b/src/qemu/qemu_monitor.h
@@ -836,6 +836,9 @@ int qemuMonitorGetSEVCapabilities(qemuMonitor *mon,
 int qemuMonitorGetSGXCapabilities(qemuMonitor *mon,
                                   virSGXCapability **capabilities);
 
+int qemuMonitorGetCCACapabilities(qemuMonitor *mon,
+                                  virCCACapability **capabilities);
+
 typedef enum {
   QEMU_MONITOR_MIGRATE_RESUME           = 1 << 0, /* resume failed post-copy migration */
   QEMU_MONITOR_MIGRATION_FLAGS_LAST
diff --git a/src/qemu/qemu_monitor_json.c b/src/qemu/qemu_monitor_json.c
index 2db38c10..cb385298 100644
--- a/src/qemu/qemu_monitor_json.c
+++ b/src/qemu/qemu_monitor_json.c
@@ -6133,6 +6133,104 @@ qemuMonitorJSONGetSGXCapabilities(qemuMonitor *mon,
 }
 
 
+static int
+qemuMonitorJSONGetCCAMeasurementAlgo(qemuMonitor *mon,
+                                     size_t *numalgo,
+                                     char ***malgo)
+{
+    g_autoptr(virJSONValue) cmd = NULL;
+    g_autoptr(virJSONValue) reply = NULL;
+    virJSONValue *caps;
+    virJSONValue *malgolist = NULL;
+    g_auto(GStrv) list = NULL;
+    size_t i;
+    size_t n = 0;
+
+    if (!(cmd = qemuMonitorJSONMakeCommand("query-cca-capabilities",
+                                           NULL)))
+        return -1;
+
+    if (qemuMonitorJSONCommand(mon, cmd, &reply) < 0)
+        return -1;
+
+    /* If the 'query-cca-capabilities' QMP command was not available
+     * we simply successfully return zero capabilities.
+     * This is the current QEMU (=9.1.91) and all non-ARM architectures */
+    if (qemuMonitorJSONHasError(reply, "CommandNotFound"))
+        return 0;
+
+    if (qemuMonitorJSONCheckError(cmd, reply) < 0)
+        return -1;
+
+    caps = virJSONValueObjectGetObject(reply, "return");
+
+    if (!(caps = qemuMonitorJSONGetReply(cmd, reply, VIR_JSON_TYPE_OBJECT)))
+        return -1;
+
+    if ((malgolist = virJSONValueObjectGetArray(caps, "sections"))) {
+        n = virJSONValueArraySize(malgolist);
+
+        /* If the received array is empty, an error is returned. */
+        if (n == 0)
+            return -1;
+
+        list = g_new0(char *, n + 1);
+
+        for (i = 0; i < n; i++) {
+            virJSONValue *cap = virJSONValueArrayGet(malgolist, i);
+            const char *measurement_algo = NULL;
+
+            if (!cap || virJSONValueGetType(cap) != VIR_JSON_TYPE_OBJECT) {
+                virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                               _("missing entry in CCA capabilities list"));
+                return -1;
+            }
+
+            if (!(measurement_algo = virJSONValueObjectGetString(cap, "measurement-algo"))) {
+                virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                           _("query-cca-capabilities reply was missing 'measurement-algo' field"));
+                return -1;
+            }
+
+            list[i] = g_strdup(measurement_algo);
+        }
+    }
+
+    *numalgo = n;
+    *malgo = g_steal_pointer(&list);
+    return 1;
+}
+
+
+/**
+ * qemuMonitorJSONGetCCACapabilities:
+ * @mon: qemu monitor object
+ * @capabilities: pointer to pointer to a CCA capability structure to be filled
+ *
+ * Returns -1 on error, 0 if CCA is not supported, and 1 if CCA is supported on
+ * the platform.
+ */
+int
+qemuMonitorJSONGetCCACapabilities(qemuMonitor *mon,
+                                  virCCACapability **capabilities)
+{
+    g_autoptr(virCCACapability) capability = NULL;
+    int ret = 0;
+
+    *capabilities = NULL;
+    capability = g_new0(virCCACapability, 1);
+
+    ret = qemuMonitorJSONGetCCAMeasurementAlgo(mon,
+                                               &capability->nCcaMeasurementAlgo,
+                                               &capability->ccaMeasurementAlgo);
+
+    if (ret > 0)
+        *capabilities = g_steal_pointer(&capability);
+
+    return ret;
+}
+
+
 static virJSONValue *
 qemuMonitorJSONBuildInetSocketAddress(const char *host,
                                       const char *port)
diff --git a/src/qemu/qemu_monitor_json.h b/src/qemu/qemu_monitor_json.h
index 921dd34e..a522beb2 100644
--- a/src/qemu/qemu_monitor_json.h
+++ b/src/qemu/qemu_monitor_json.h
@@ -168,6 +168,10 @@ int
 qemuMonitorJSONGetSEVCapabilities(qemuMonitor *mon,
                                   virSEVCapability **capabilities);
 
+int
+qemuMonitorJSONGetCCACapabilities(qemuMonitor *mon,
+                                  virCCACapability **capabilities);
+
 int
 qemuMonitorJSONMigrate(qemuMonitor *mon,
                        unsigned int flags,
diff --git a/tests/domaincapsdata/bhyve_basic.x86_64.xml b/tests/domaincapsdata/bhyve_basic.x86_64.xml
index dd054577..415c10ea 100644
--- a/tests/domaincapsdata/bhyve_basic.x86_64.xml
+++ b/tests/domaincapsdata/bhyve_basic.x86_64.xml
@@ -34,5 +34,6 @@
     <genid supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/bhyve_fbuf.x86_64.xml b/tests/domaincapsdata/bhyve_fbuf.x86_64.xml
index 0b1d9c17..68bfd71f 100644
--- a/tests/domaincapsdata/bhyve_fbuf.x86_64.xml
+++ b/tests/domaincapsdata/bhyve_fbuf.x86_64.xml
@@ -51,5 +51,6 @@
     <genid supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/bhyve_uefi.x86_64.xml b/tests/domaincapsdata/bhyve_uefi.x86_64.xml
index 69fff197..42be6674 100644
--- a/tests/domaincapsdata/bhyve_uefi.x86_64.xml
+++ b/tests/domaincapsdata/bhyve_uefi.x86_64.xml
@@ -43,5 +43,6 @@
     <genid supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/empty.xml b/tests/domaincapsdata/empty.xml
index 97752ca0..5157d684 100644
--- a/tests/domaincapsdata/empty.xml
+++ b/tests/domaincapsdata/empty.xml
@@ -14,5 +14,6 @@
   <features>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/libxl-xenfv.xml b/tests/domaincapsdata/libxl-xenfv.xml
index c71d7595..e27f2372 100644
--- a/tests/domaincapsdata/libxl-xenfv.xml
+++ b/tests/domaincapsdata/libxl-xenfv.xml
@@ -77,5 +77,6 @@
     <genid supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/libxl-xenpv.xml b/tests/domaincapsdata/libxl-xenpv.xml
index 8ae2370b..8b138c16 100644
--- a/tests/domaincapsdata/libxl-xenpv.xml
+++ b/tests/domaincapsdata/libxl-xenpv.xml
@@ -67,5 +67,6 @@
     <genid supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml
index e5ffe393..912897ec 100644
--- a/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml
@@ -324,6 +324,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_4.2.0-tcg.x86_64.xml
index a849e8f1..63996a5a 100644
--- a/tests/domaincapsdata/qemu_4.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0-tcg.x86_64.xml
@@ -269,6 +269,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_4.2.0-virt.aarch64.xml
index a7c2e0ba..727294ce 100644
--- a/tests/domaincapsdata/qemu_4.2.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0-virt.aarch64.xml
@@ -201,6 +201,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0.aarch64.xml b/tests/domaincapsdata/qemu_4.2.0.aarch64.xml
index a7c2e0ba..727294ce 100644
--- a/tests/domaincapsdata/qemu_4.2.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0.aarch64.xml
@@ -201,6 +201,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0.ppc64.xml b/tests/domaincapsdata/qemu_4.2.0.ppc64.xml
index 2c41777e..b42fd490 100644
--- a/tests/domaincapsdata/qemu_4.2.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0.ppc64.xml
@@ -169,6 +169,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0.s390x.xml b/tests/domaincapsdata/qemu_4.2.0.s390x.xml
index 809ce9c9..51f47657 100644
--- a/tests/domaincapsdata/qemu_4.2.0.s390x.xml
+++ b/tests/domaincapsdata/qemu_4.2.0.s390x.xml
@@ -275,6 +275,7 @@
     <s390-pv supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_4.2.0.x86_64.xml b/tests/domaincapsdata/qemu_4.2.0.x86_64.xml
index 32ba46eb..2c78cb81 100644
--- a/tests/domaincapsdata/qemu_4.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0.x86_64.xml
@@ -324,6 +324,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_5.0.0-q35.x86_64.xml
index 70bd7bc4..7cc17456 100644
--- a/tests/domaincapsdata/qemu_5.0.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0-q35.x86_64.xml
@@ -326,6 +326,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0-tcg-virt.riscv64.xml b/tests/domaincapsdata/qemu_5.0.0-tcg-virt.riscv64.xml
index c487d467..ce4f0f9f 100644
--- a/tests/domaincapsdata/qemu_5.0.0-tcg-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0-tcg-virt.riscv64.xml
@@ -154,6 +154,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_5.0.0-tcg.x86_64.xml
index d69ea76a..5bc87668 100644
--- a/tests/domaincapsdata/qemu_5.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0-tcg.x86_64.xml
@@ -271,6 +271,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_5.0.0-virt.aarch64.xml
index 2466dda7..a7955432 100644
--- a/tests/domaincapsdata/qemu_5.0.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0-virt.aarch64.xml
@@ -214,6 +214,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0-virt.riscv64.xml b/tests/domaincapsdata/qemu_5.0.0-virt.riscv64.xml
index b0e4aafc..09617e26 100644
--- a/tests/domaincapsdata/qemu_5.0.0-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0-virt.riscv64.xml
@@ -157,6 +157,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0.aarch64.xml b/tests/domaincapsdata/qemu_5.0.0.aarch64.xml
index 2466dda7..a7955432 100644
--- a/tests/domaincapsdata/qemu_5.0.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0.aarch64.xml
@@ -214,6 +214,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0.ppc64.xml b/tests/domaincapsdata/qemu_5.0.0.ppc64.xml
index c66d0645..84928251 100644
--- a/tests/domaincapsdata/qemu_5.0.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0.ppc64.xml
@@ -176,6 +176,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.0.0.x86_64.xml b/tests/domaincapsdata/qemu_5.0.0.x86_64.xml
index c57d4f32..ca788d3c 100644
--- a/tests/domaincapsdata/qemu_5.0.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.0.0.x86_64.xml
@@ -326,6 +326,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.1.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_5.1.0-q35.x86_64.xml
index f276f0b7..2b8efb9e 100644
--- a/tests/domaincapsdata/qemu_5.1.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.1.0-q35.x86_64.xml
@@ -258,6 +258,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.1.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_5.1.0-tcg.x86_64.xml
index eea710a0..6b49c06d 100644
--- a/tests/domaincapsdata/qemu_5.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.1.0-tcg.x86_64.xml
@@ -271,6 +271,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.1.0.sparc.xml b/tests/domaincapsdata/qemu_5.1.0.sparc.xml
index a471a937..7e5eb5df 100644
--- a/tests/domaincapsdata/qemu_5.1.0.sparc.xml
+++ b/tests/domaincapsdata/qemu_5.1.0.sparc.xml
@@ -140,6 +140,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.1.0.x86_64.xml b/tests/domaincapsdata/qemu_5.1.0.x86_64.xml
index 56e30cc3..17576e30 100644
--- a/tests/domaincapsdata/qemu_5.1.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.1.0.x86_64.xml
@@ -258,6 +258,7 @@
     <async-teardown supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_5.2.0-q35.x86_64.xml
index 35bc747d..c653d765 100644
--- a/tests/domaincapsdata/qemu_5.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0-q35.x86_64.xml
@@ -259,6 +259,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0-tcg-virt.riscv64.xml b/tests/domaincapsdata/qemu_5.2.0-tcg-virt.riscv64.xml
index eaed71b4..c309a3c2 100644
--- a/tests/domaincapsdata/qemu_5.2.0-tcg-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0-tcg-virt.riscv64.xml
@@ -157,6 +157,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_5.2.0-tcg.x86_64.xml
index 33b55317..f61cdce6 100644
--- a/tests/domaincapsdata/qemu_5.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0-tcg.x86_64.xml
@@ -272,6 +272,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_5.2.0-virt.aarch64.xml
index a35dd78c..c093dde9 100644
--- a/tests/domaincapsdata/qemu_5.2.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0-virt.aarch64.xml
@@ -215,6 +215,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0-virt.riscv64.xml b/tests/domaincapsdata/qemu_5.2.0-virt.riscv64.xml
index 14d936ea..4839e7f0 100644
--- a/tests/domaincapsdata/qemu_5.2.0-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0-virt.riscv64.xml
@@ -160,6 +160,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0.aarch64.xml b/tests/domaincapsdata/qemu_5.2.0.aarch64.xml
index a35dd78c..c093dde9 100644
--- a/tests/domaincapsdata/qemu_5.2.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0.aarch64.xml
@@ -215,6 +215,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0.ppc64.xml b/tests/domaincapsdata/qemu_5.2.0.ppc64.xml
index a6fcc39c..bd1e3bfd 100644
--- a/tests/domaincapsdata/qemu_5.2.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0.ppc64.xml
@@ -177,6 +177,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0.s390x.xml b/tests/domaincapsdata/qemu_5.2.0.s390x.xml
index 084a221f..ca78cb76 100644
--- a/tests/domaincapsdata/qemu_5.2.0.s390x.xml
+++ b/tests/domaincapsdata/qemu_5.2.0.s390x.xml
@@ -278,6 +278,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_5.2.0.x86_64.xml b/tests/domaincapsdata/qemu_5.2.0.x86_64.xml
index 2889fce1..c960d52b 100644
--- a/tests/domaincapsdata/qemu_5.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_5.2.0.x86_64.xml
@@ -259,6 +259,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.0.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_6.0.0-q35.x86_64.xml
index 5bc2bdd1..a4367817 100644
--- a/tests/domaincapsdata/qemu_6.0.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0-q35.x86_64.xml
@@ -266,6 +266,7 @@
       <maxESGuests>450</maxESGuests>
     </sev>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>sev</value>
diff --git a/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml
index a39f777e..3d414601 100644
--- a/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0-tcg.x86_64.xml
@@ -280,6 +280,7 @@
       <maxESGuests>450</maxESGuests>
     </sev>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>sev</value>
diff --git a/tests/domaincapsdata/qemu_6.0.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_6.0.0-virt.aarch64.xml
index 780436ce..152682fe 100644
--- a/tests/domaincapsdata/qemu_6.0.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0-virt.aarch64.xml
@@ -217,6 +217,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.0.0.aarch64.xml b/tests/domaincapsdata/qemu_6.0.0.aarch64.xml
index 780436ce..152682fe 100644
--- a/tests/domaincapsdata/qemu_6.0.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0.aarch64.xml
@@ -217,6 +217,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.0.0.s390x.xml b/tests/domaincapsdata/qemu_6.0.0.s390x.xml
index 2bc02dd4..ef712c09 100644
--- a/tests/domaincapsdata/qemu_6.0.0.s390x.xml
+++ b/tests/domaincapsdata/qemu_6.0.0.s390x.xml
@@ -279,6 +279,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>s390-pv</value>
diff --git a/tests/domaincapsdata/qemu_6.0.0.x86_64.xml b/tests/domaincapsdata/qemu_6.0.0.x86_64.xml
index 0c322b11..1d27c012 100644
--- a/tests/domaincapsdata/qemu_6.0.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.0.0.x86_64.xml
@@ -266,6 +266,7 @@
       <maxESGuests>450</maxESGuests>
     </sev>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>sev</value>
diff --git a/tests/domaincapsdata/qemu_6.1.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_6.1.0-q35.x86_64.xml
index b5f488d5..5f64467a 100644
--- a/tests/domaincapsdata/qemu_6.1.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.1.0-q35.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml
index 3137894a..ab50f631 100644
--- a/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.1.0-tcg.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.1.0.x86_64.xml b/tests/domaincapsdata/qemu_6.1.0.x86_64.xml
index a99dccf8..49f13ed4 100644
--- a/tests/domaincapsdata/qemu_6.1.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.1.0.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_6.2.0-q35.x86_64.xml
index 2326871d..fd0bdefc 100644
--- a/tests/domaincapsdata/qemu_6.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0-q35.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml
index 4270ba0b..820e7c08 100644
--- a/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0-tcg.x86_64.xml
@@ -277,6 +277,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_6.2.0-virt.aarch64.xml
index d0903aac..44868113 100644
--- a/tests/domaincapsdata/qemu_6.2.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0-virt.aarch64.xml
@@ -224,6 +224,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0.aarch64.xml b/tests/domaincapsdata/qemu_6.2.0.aarch64.xml
index d0903aac..44868113 100644
--- a/tests/domaincapsdata/qemu_6.2.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0.aarch64.xml
@@ -224,6 +224,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0.ppc64.xml b/tests/domaincapsdata/qemu_6.2.0.ppc64.xml
index fe62364d..7a2f6b5e 100644
--- a/tests/domaincapsdata/qemu_6.2.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0.ppc64.xml
@@ -176,6 +176,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_6.2.0.x86_64.xml b/tests/domaincapsdata/qemu_6.2.0.x86_64.xml
index 63ddcaec..3357dc26 100644
--- a/tests/domaincapsdata/qemu_6.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_6.2.0.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0-hvf.aarch64+hvf.xml b/tests/domaincapsdata/qemu_7.0.0-hvf.aarch64+hvf.xml
index 32eee7ea..d4b5ef30 100644
--- a/tests/domaincapsdata/qemu_7.0.0-hvf.aarch64+hvf.xml
+++ b/tests/domaincapsdata/qemu_7.0.0-hvf.aarch64+hvf.xml
@@ -178,6 +178,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_7.0.0-q35.x86_64.xml
index 83c62a11..5aa10bed 100644
--- a/tests/domaincapsdata/qemu_7.0.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0-q35.x86_64.xml
@@ -272,6 +272,7 @@
         <section node='1' size='262144' unit='KiB'/>
       </sections>
     </sgx>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_7.0.0-tcg.x86_64.xml
index 00adacd1..6b00d2a2 100644
--- a/tests/domaincapsdata/qemu_7.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0-tcg.x86_64.xml
@@ -287,6 +287,7 @@
         <section node='1' size='262144' unit='KiB'/>
       </sections>
     </sgx>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_7.0.0-virt.aarch64.xml
index 216e3999..661c8160 100644
--- a/tests/domaincapsdata/qemu_7.0.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0-virt.aarch64.xml
@@ -223,6 +223,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0.aarch64.xml b/tests/domaincapsdata/qemu_7.0.0.aarch64.xml
index 216e3999..661c8160 100644
--- a/tests/domaincapsdata/qemu_7.0.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0.aarch64.xml
@@ -223,6 +223,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0.ppc64.xml b/tests/domaincapsdata/qemu_7.0.0.ppc64.xml
index e0fda8fc..544662cb 100644
--- a/tests/domaincapsdata/qemu_7.0.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0.ppc64.xml
@@ -181,6 +181,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.0.0.x86_64.xml b/tests/domaincapsdata/qemu_7.0.0.x86_64.xml
index 97e38888..d24320a9 100644
--- a/tests/domaincapsdata/qemu_7.0.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.0.0.x86_64.xml
@@ -272,6 +272,7 @@
         <section node='1' size='262144' unit='KiB'/>
       </sections>
     </sgx>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.1.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_7.1.0-q35.x86_64.xml
index 1b55b9b4..8cbc94b5 100644
--- a/tests/domaincapsdata/qemu_7.1.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.1.0-q35.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.1.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_7.1.0-tcg.x86_64.xml
index 37697928..64d7433c 100644
--- a/tests/domaincapsdata/qemu_7.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.1.0-tcg.x86_64.xml
@@ -276,6 +276,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.1.0.ppc64.xml b/tests/domaincapsdata/qemu_7.1.0.ppc64.xml
index bbdf8a60..5e770f07 100644
--- a/tests/domaincapsdata/qemu_7.1.0.ppc64.xml
+++ b/tests/domaincapsdata/qemu_7.1.0.ppc64.xml
@@ -174,6 +174,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.1.0.x86_64.xml b/tests/domaincapsdata/qemu_7.1.0.x86_64.xml
index 9d5ff18c..d839197d 100644
--- a/tests/domaincapsdata/qemu_7.1.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.1.0.x86_64.xml
@@ -262,6 +262,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0-hvf.x86_64+hvf.xml b/tests/domaincapsdata/qemu_7.2.0-hvf.x86_64+hvf.xml
index 091735d8..13385e46 100644
--- a/tests/domaincapsdata/qemu_7.2.0-hvf.x86_64+hvf.xml
+++ b/tests/domaincapsdata/qemu_7.2.0-hvf.x86_64+hvf.xml
@@ -186,6 +186,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_7.2.0-q35.x86_64.xml
index 73475e49..2d407c92 100644
--- a/tests/domaincapsdata/qemu_7.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.2.0-q35.x86_64.xml
@@ -269,6 +269,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64+hvf.xml b/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64+hvf.xml
index 38fedd6c..a8d7816d 100644
--- a/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64+hvf.xml
+++ b/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64+hvf.xml
@@ -267,6 +267,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64.xml
index 38fedd6c..a8d7816d 100644
--- a/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.2.0-tcg.x86_64.xml
@@ -267,6 +267,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0.ppc.xml b/tests/domaincapsdata/qemu_7.2.0.ppc.xml
index 416ebc6f..b4908ce2 100644
--- a/tests/domaincapsdata/qemu_7.2.0.ppc.xml
+++ b/tests/domaincapsdata/qemu_7.2.0.ppc.xml
@@ -161,6 +161,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_7.2.0.x86_64.xml b/tests/domaincapsdata/qemu_7.2.0.x86_64.xml
index b8b2aada..58f333fc 100644
--- a/tests/domaincapsdata/qemu_7.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_7.2.0.x86_64.xml
@@ -269,6 +269,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.0.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_8.0.0-q35.x86_64.xml
index c7c118d5..d70e32c0 100644
--- a/tests/domaincapsdata/qemu_8.0.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.0.0-q35.x86_64.xml
@@ -270,6 +270,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.0.0-tcg-virt.riscv64.xml b/tests/domaincapsdata/qemu_8.0.0-tcg-virt.riscv64.xml
index af787b8b..7da008be 100644
--- a/tests/domaincapsdata/qemu_8.0.0-tcg-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_8.0.0-tcg-virt.riscv64.xml
@@ -164,6 +164,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.0.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_8.0.0-tcg.x86_64.xml
index ab81714c..2e8f86bc 100644
--- a/tests/domaincapsdata/qemu_8.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.0.0-tcg.x86_64.xml
@@ -272,6 +272,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.0.0-virt.riscv64.xml b/tests/domaincapsdata/qemu_8.0.0-virt.riscv64.xml
index 0e3e9b81..81a2855c 100644
--- a/tests/domaincapsdata/qemu_8.0.0-virt.riscv64.xml
+++ b/tests/domaincapsdata/qemu_8.0.0-virt.riscv64.xml
@@ -167,6 +167,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.0.0.x86_64.xml b/tests/domaincapsdata/qemu_8.0.0.x86_64.xml
index ad9eaf54..895fb133 100644
--- a/tests/domaincapsdata/qemu_8.0.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.0.0.x86_64.xml
@@ -270,6 +270,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.1.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_8.1.0-q35.x86_64.xml
index aa01168a..9f3fdcae 100644
--- a/tests/domaincapsdata/qemu_8.1.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.1.0-q35.x86_64.xml
@@ -273,6 +273,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.1.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_8.1.0-tcg.x86_64.xml
index 8e33506a..c4c1e3d1 100644
--- a/tests/domaincapsdata/qemu_8.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.1.0-tcg.x86_64.xml
@@ -274,6 +274,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.1.0.s390x.xml b/tests/domaincapsdata/qemu_8.1.0.s390x.xml
index 92fa89f0..e70e75e2 100644
--- a/tests/domaincapsdata/qemu_8.1.0.s390x.xml
+++ b/tests/domaincapsdata/qemu_8.1.0.s390x.xml
@@ -275,6 +275,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>s390-pv</value>
diff --git a/tests/domaincapsdata/qemu_8.1.0.x86_64.xml b/tests/domaincapsdata/qemu_8.1.0.x86_64.xml
index 532a9b2d..57b7eea8 100644
--- a/tests/domaincapsdata/qemu_8.1.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.1.0.x86_64.xml
@@ -273,6 +273,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.2.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_8.2.0-q35.x86_64.xml
index cebc4e01..cce8fcaf 100644
--- a/tests/domaincapsdata/qemu_8.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0-q35.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.2.0-tcg-virt.loongarch64.xml b/tests/domaincapsdata/qemu_8.2.0-tcg-virt.loongarch64.xml
index 3917ce61..728ed3f9 100644
--- a/tests/domaincapsdata/qemu_8.2.0-tcg-virt.loongarch64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0-tcg-virt.loongarch64.xml
@@ -168,6 +168,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.2.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_8.2.0-tcg.x86_64.xml
index a68b8a6f..16dddea1 100644
--- a/tests/domaincapsdata/qemu_8.2.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0-tcg.x86_64.xml
@@ -274,6 +274,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_8.2.0-virt.aarch64.xml b/tests/domaincapsdata/qemu_8.2.0-virt.aarch64.xml
index 2f661b7e..9013e527 100644
--- a/tests/domaincapsdata/qemu_8.2.0-virt.aarch64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0-virt.aarch64.xml
@@ -230,6 +230,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.2.0-virt.loongarch64.xml b/tests/domaincapsdata/qemu_8.2.0-virt.loongarch64.xml
index edddb6d8..55468c65 100644
--- a/tests/domaincapsdata/qemu_8.2.0-virt.loongarch64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0-virt.loongarch64.xml
@@ -172,6 +172,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.2.0.aarch64.xml b/tests/domaincapsdata/qemu_8.2.0.aarch64.xml
index 2f661b7e..9013e527 100644
--- a/tests/domaincapsdata/qemu_8.2.0.aarch64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0.aarch64.xml
@@ -230,6 +230,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.2.0.armv7l.xml b/tests/domaincapsdata/qemu_8.2.0.armv7l.xml
index fb0f7665..862ea3b4 100644
--- a/tests/domaincapsdata/qemu_8.2.0.armv7l.xml
+++ b/tests/domaincapsdata/qemu_8.2.0.armv7l.xml
@@ -178,6 +178,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_8.2.0.s390x.xml b/tests/domaincapsdata/qemu_8.2.0.s390x.xml
index cc8bbec0..0d1d54f8 100644
--- a/tests/domaincapsdata/qemu_8.2.0.s390x.xml
+++ b/tests/domaincapsdata/qemu_8.2.0.s390x.xml
@@ -275,6 +275,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='yes'>
       <enum name='sectype'>
         <value>s390-pv</value>
diff --git a/tests/domaincapsdata/qemu_8.2.0.x86_64.xml b/tests/domaincapsdata/qemu_8.2.0.x86_64.xml
index 0bd89504..2918526e 100644
--- a/tests/domaincapsdata/qemu_8.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_8.2.0.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.0.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_9.0.0-q35.x86_64.xml
index b9a21549..b614b6b7 100644
--- a/tests/domaincapsdata/qemu_9.0.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.0.0-q35.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.0.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_9.0.0-tcg.x86_64.xml
index 25bf1d84..f917b160 100644
--- a/tests/domaincapsdata/qemu_9.0.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.0.0-tcg.x86_64.xml
@@ -276,6 +276,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.0.0.sparc.xml b/tests/domaincapsdata/qemu_9.0.0.sparc.xml
index ab5c3dc4..0cca40dd 100644
--- a/tests/domaincapsdata/qemu_9.0.0.sparc.xml
+++ b/tests/domaincapsdata/qemu_9.0.0.sparc.xml
@@ -140,6 +140,7 @@
     <ps2 supported='no'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <launchSecurity supported='no'/>
   </features>
 </domainCapabilities>
diff --git a/tests/domaincapsdata/qemu_9.0.0.x86_64.xml b/tests/domaincapsdata/qemu_9.0.0.x86_64.xml
index f785e83f..e0d0220c 100644
--- a/tests/domaincapsdata/qemu_9.0.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.0.0.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.1.0-q35.x86_64.xml b/tests/domaincapsdata/qemu_9.1.0-q35.x86_64.xml
index ef364621..14f6898c 100644
--- a/tests/domaincapsdata/qemu_9.1.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.1.0-q35.x86_64.xml
@@ -279,6 +279,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.1.0-tcg.x86_64.xml b/tests/domaincapsdata/qemu_9.1.0-tcg.x86_64.xml
index 2415f76f..be36933a 100644
--- a/tests/domaincapsdata/qemu_9.1.0-tcg.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.1.0-tcg.x86_64.xml
@@ -275,6 +275,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
diff --git a/tests/domaincapsdata/qemu_9.1.0.x86_64.xml b/tests/domaincapsdata/qemu_9.1.0.x86_64.xml
index 8cba35fb..9c669143 100644
--- a/tests/domaincapsdata/qemu_9.1.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_9.1.0.x86_64.xml
@@ -279,6 +279,7 @@
     <ps2 supported='yes'/>
     <sev supported='no'/>
     <sgx supported='no'/>
+    <cca supported='no'/>
     <hyperv supported='yes'>
       <enum name='features'>
         <value>relaxed</value>
-- 
2.51.0

